lib/events.js

```diff
@@ -621,22 +629,46 @@ function unwrapListeners(arr) {
   return ret;
 }
 
-function once(emitter, name) {
+function once(emitter, name, options = {}) {
+  const { signal } = { ...options };
+  if (signal !== undefined &&
+      (signal === null ||
+       typeof signal !== 'object' ||
+       !('aborted' in signal))) {
```

Comment:
Could we extract this in some sort of validator?

Reply:
Yes, I plan to